core/src/jvmTest/kotlin/com/ustadmobile/core/controller/PersonAccountEditPresenterTest.kt

```diff
@@ -0,0 +1,218 @@
+        val presenter = PersonAccountEditPresenter(context,args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<Boolean>{
+            verify(mockView, timeout(defaultTimeOut).atLeastOnce()).firstPasswordFieldRequiredErrorVisible = capture()
+            assertEquals("Current password field error was shown", true, firstValue)
+        }
+    }
+
+    @Test
+    fun givenPresenterCreated_whenNewPasswordIsNotFilledAndSaveClicked_thenShouldShowError(){
+        val person = createPerson(isAdmin = true)
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString())
+        val presenter = PersonAccountEditPresenter(context, args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<Boolean>{
+            verify(mockView, timeout(defaultTimeOut).atLeastOnce()).secondPasswordFieldRequiredErrorVisible = capture()
+            assertEquals("Password field error was shown", true, firstValue)
+        }
+    }
+
+    @Test
+    fun givenPresenterCreatedInPasswordResetMode_whenAllFieldsAreFilledAndSaveClicked_thenShouldChangePassword(){
+        enQueuePasswordChangeResponse()
+
+        mockView = mock{
+            on{firstPassword}.thenReturn("oldPassword")
+            on{secondPassword}.thenReturn("password")
+        }
+        val person = createPerson(isAdmin = true, withUsername = true)
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString(),
+                UstadView.ARG_SERVER_URL to serverUrl)
+        val presenter = PersonAccountEditPresenter(context, args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        presenter.handleClickSave(person)
+        argumentCaptor<String>{
+            verifyBlocking(accountManager, timeout(defaultTimeOut).atLeastOnce()){
+                changePassword(capture(), any(), any(), any())
+                assertEquals("Password change task was started", person.username, firstValue)
+            }
+        }
+        verify(mockView, timeout(defaultTimeOut)).finishWithResult(any())
+    }
+
+
+    @Test
+    fun givenPersonAccountInAccountCreationMode_whenAllFieldsAreFilledAndSaveClicked_thenShouldCreateAnAccount(){
+        enQueuePasswordChangeResponse()
+
+        mockView = mock{
+            on{firstPassword}.thenReturn("password")
+            on{secondPassword}.thenReturn("password")
+        }
+        val person = createPerson(isAdmin = true, withUsername = false)
+        val args = mapOf(UstadView.ARG_ENTITY_UID to person.personUid.toString(),
+                UstadView.ARG_SERVER_URL to serverUrl)
+        val presenter = PersonAccountEditPresenter(context, args,mockView, di, mockLifecycleOwner)
+        presenter.onCreate(null)
+        //wait for onLoadEntityFromDb to be called
+        sleep(1000)
```

Comment:
We shouldn't use sleep as this would normally wait to long and sometimes not long enough. Even 1 second on every test will add up. You can use : captureLastEntityValue(timeout) ( which can be found in UstadSingleEntityView). 

Because the view is a mock, you can wait for the entity object to be set. E.g. this would also work:

```
verify(mockView, timeout(5000)).entity = any()
```